File: BaekJoon/Silver/18110/C++/18110.cpp
// no.18110: solved.ac (S4)

#include <cstdio>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;

int main() {
    int n;
    scanf("%d", &n);
    vector<int> comments;
    for(int i=0; i<n; i++) {
        int diff;
        scanf("%d", &diff);
        comments.push_back(diff);
    }
    if(n==0) {
        printf("0\n");
        return 0;
    }
    sort(comments.begin(), comments.end());
    int cut = (int)floor(comments.size() * 0.15 + 0.5);
    int sum = 0;
    for(int i=cut; i<n-cut; i++) {
        sum += comments[i];
    }
    printf("%d\n", (int)floor((double)sum/(n-cut*2)+0.5));
    return 0; 
}

File: BaekJoon/Silver/18110/Python/18110.py
import sys
import math

n = int(sys.stdin.readline())
comments = [int(sys.stdin.readline()) for i in range(n)]
comments.sort()
deleted_range = math.floor(len(comments)*0.15+0.5)
comments = comments[deleted_range:len(comments)-deleted_range]
print(math.floor(sum(comments)/len(comments)+0.5) if n != 0 else 0)
